src/hooks/useNotificateTxState.tsx

```diff
@@ -0,0 +1,50 @@
+import { GetBlockNumberErrorType, Hash } from 'viem';
+import { useAccount } from 'wagmi';
+import { useSetNotification } from '~/hooks/useNotification';
+
+const pendingMessage = 'Pending Transaction',
+  successMessage = 'Transaction successful',
+  errorMessage = 'Error while processing transaction';
+
+const defaultBlockExplorer = '';
+
+export const useNotificateTxState = () => {
+  const setNotification = useSetNotification();
+  const { chain } = useAccount();
+
+  const blockExplorer = chain?.blockExplorers?.default.url || defaultBlockExplorer;
+
+  const notificateTxPending = (hash: Hash, message: string = pendingMessage) => {
+    setNotification({
+      type: 'loading',
+      message,
+      link: {
+        href: `${blockExplorer}/tx/${hash}`,
+        text: 'See transaction',
+      },
+      timeout: 0,
+    });
+  };
+
+  const notificateTxSuccess = (hash: Hash, message: string = successMessage) => {
+    setNotification({
+      type: 'success',
+      message,
+      link: {
+        href: `${blockExplorer}/tx/${hash}`,
+        text: 'See transaction',
+      },
+      timeout: 0,
+    });
+  };
+
+  const notificateTxError = (error: unknown, message: string = errorMessage) => {
+    setNotification({
+      type: 'error',
+      message: `${message}. Error: ` + (error as GetBlockNumberErrorType)?.name,
+      timeout: 0,
+    });
+  };
```

Comment:
i would recommend to abstract this logic to one function to avoid repeat to much code

Reply:
wdyt?

```jsx
const notificateTx = (args: NotificationTxStateArgs) => {
  let notificationMessage = args.message;
  let link;

  switch (args.type) {
    case 'loading':
      notificationMessage = args.message || pendingMessage;
      link = {
        href: `${blockExplorer}/tx/${args.hash}`,
        text: 'See transaction',
      };
      break;
    case 'success':
      notificationMessage = args.message || successMessage;
      link = {
        href: `${blockExplorer}/tx/${args.hash}`,
        text: 'See transaction',
      };
      break;
    case 'error':
      notificationMessage =
        `${args.message || errorMessage}. Error: ` + (args.error as GetBlockNumberErrorType)?.name;
      break;
  }

  setNotification({
    type: args.type,
    message: notificationMessage,
    link,
    timeout: 0,
  });
};
```

Reply:
yes! thats better. Also if you want more flexibility you can something like:

```ts

const notificateTxPending = (hash: Hash, message: string = pendingMessage) => {
  setNotification({
    type: getNotificationType(message),
    message: getCustomMessage(message),
    link: getNotificationLink(hash, message),
    timeot: 0,	
  })
}
```

Reply:
https://github.com/0xShishigami/feo-web3-react-boilerplate/pull/7/commits/9497d06e1cafa4cfeb2133f65c4d5f6b4c2daac8